lightning/src/ln/channelmanager.rs

```diff
@@ -10644,36 +10644,41 @@ where
 				}
 			},
 			OffersMessage::Invoice(invoice) => {
-				let expected_payment_id = match context {
+				let payer_data = match context {
 					OffersContext::Unknown {} if invoice.is_for_refund_without_paths() => None,
-					OffersContext::OutboundPayment { payment_id, .. } => Some(payment_id),
+					OffersContext::OutboundPayment { payment_id, nonce } => Some((payment_id, nonce)),
 					_ => return ResponseInstruction::NoResponse,
 				};
 
-				let result = match invoice.verify(expanded_key, secp_ctx) {
-					Ok(payment_id) => {
-						if let Some(expected_payment_id) = expected_payment_id {
-							if payment_id != expected_payment_id {
-								return ResponseInstruction::NoResponse;
-							}
```

Comment:
This check gets removed after being added in 6c180de7fb4a9e27aef4fdaf64a993fee1d5717b, is that intentional?